File: backend/internal/idgen/idgen.go
package idgen

import (
	"crypto/rand"
	"encoding/base64"
	"encoding/hex"
	"fmt"
)

// GenerateID generates an ID with prefix in the format: prefix_<8_random_hex_bytes>
func GenerateID(prefix string) string {
	b := make([]byte, 8)
	if _, err := rand.Read(b); err != nil {
		panic(fmt.Sprintf("生成随机 ID 失败: %v", err))
	}
	return prefix + "_" + hex.EncodeToString(b)
}

// GenerateAPIKey generates an API key in the format: sk-ant-api03-<40_random_bytes_base64url>
// (Claude API format compatible)
func GenerateAPIKey() string {
	b := make([]byte, 40)
	if _, err := rand.Read(b); err != nil {
		panic(fmt.Sprintf("生成 API Key 失败: %v", err))
	}
	// base64url without padding
	encoded := base64.RawURLEncoding.EncodeToString(b)
	return "sk-ant-api03-" + encoded
}

File: backend-java/src/main/java/com/copilotmanager/idgen/IdGen.java
package com.copilotmanager.idgen;

import java.security.SecureRandom;
import java.util.Base64;

public final class IdGen {

    private static final SecureRandom RANDOM = new SecureRandom();

    private IdGen() {}

    public static String generateId(String prefix) {
        byte[] bytes = new byte[8];
        RANDOM.nextBytes(bytes);
        StringBuilder sb = new StringBuilder(bytes.length * 2);
        for (byte b : bytes) sb.append(String.format("%02x", b));
        return prefix + "_" + sb;
    }

    public static String generateApiKey() {
        // sk-ant-api03-<40 bytes base64url>
        byte[] bytes = new byte[40];
        RANDOM.nextBytes(bytes);
        String b64 = Base64.getUrlEncoder().withoutPadding().encodeToString(bytes);
        return "sk-ant-api03-" + b64;
    }

    public static String generateSessionId() {
        byte[] bytes = new byte[16];
        RANDOM.nextBytes(bytes);
        StringBuilder sb = new StringBuilder(32);
        for (byte b : bytes) sb.append(String.format("%02x", b));
        return sb.toString();
    }
}
